src/work-allocation/resolvers/location-resolver.service.ts

```diff
@@ -83,14 +83,18 @@ export class LocationResolver implements Resolve<LocationModel[]> {
       }
       if (roleJurisdiction && roleAssignment.roleType === 'ORGANISATION'
         && roleAssignment.primaryLocation && roleAssignment.substantive.toLocaleLowerCase() === 'y') {
-        if (!locations.find((location) => location.id === roleAssignment.primaryLocation)) {
+        if (!locations.find((location) => location.id === roleAssignment.primaryLocation && location.services.includes(roleJurisdiction))) {
```

Comment:
`roleAssignment.primaryLocation && location.services && location.services.includes(roleJurisdiction)`

Reply:
@udaydenduluri33  FYI. Primary location is inserted many times in roleAssignments and same Primary Location exists for different services. So, ignoring duplicates